### PR TITLE
API for server powerlimit

### PR DESCRIPTION
**Pull request by PremchandJujjuri**
Developed api's to set the power limit correction in ms & exceptions to handle after reach power limit

### CODE REVIEW

File: lib/redfish_api.py [hunk 1]
```diff
@@ -704,6 +704,53 @@ def get_storage_inventory(self) -> (bool, list):
             ##traceback.print_exc()
             logger.error("error_msg: {}".format(e))
             return False, storage_details
+    
+    def set_power_limit_correction_time(self,power_limit_correction_time=200):
```
Comment (Deepti1289) [inline lib/redfish_api.py:708]: please provide doc string

**Review by Deepti1289 — APPROVED** [review]

**Comment by Deepti1289** [comment]
Code looks good

File: lib/redfish_api.py [hunk 2]
```diff
@@ -704,6 +704,53 @@ def get_storage_inventory(self) -> (bool, list):
...
+            logger.error("error msg: {}".format(e))
+            return False,None
+
+    def set_power_limit_exception(self, power_limit_exception="HardPowerOff"):
+        try:
+            power_url=self.config_dict["power_url"]
+            logger.info(power_url)
+            bool_resp1,response_power_url=self.redfish_response("get",power_url)
+            #logger.info(response_power_url)
+            if bool_resp1==True:
```
Comment (sathishpos) [inline lib/redfish_api.py:756]: no code defined in if bool_resp1==True: block

File: lib/redfish_api.py [hunk 3]
```diff
@@ -704,6 +704,53 @@ def get_storage_inventory(self) -> (bool, list):
...
+                else:
+                    raise Exception("Failed")
+            else:
+                raise Exception("Failed")
+        except Exception as e:
+            #traceback.print_exc()
+            logger.error("error msg: {}".format(e))
+            return False,None
+
+    def set_power_limit_exception(self, power_limit_exception="HardPowerOff"):
```
Comment (deeptis13) [inline lib/redfish_api.py:745]: Please provide the doc string

**Review by deeptis13 — APPROVED** [review]

File: lib/redfish_api.py (hunk 3 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py:745]: Addressed the review comments

File: lib/redfish_api.py (hunk 2 above)
Comment (PremchandJujjuri) [reply lib/redfish_api.py:756]: Addressed the review comment by adding the logic to fetch the etag info from the @odata

**Review by sathishpos — APPROVED** [review]

**Pull request merged**